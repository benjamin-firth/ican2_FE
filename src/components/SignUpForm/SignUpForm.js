import React from 'react';
import './SignUpForm.scss';

const SignUpForm = () => {

  return (
    <form>
      <p>This the sign up form.</p>
    </form>
  );
}

export default SignUpForm;
