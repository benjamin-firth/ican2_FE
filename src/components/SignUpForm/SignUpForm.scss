@import '../../variables.scss';

.sign-up-container {
  background-color: $primarycolor;
  display: flex;
  flex-direction: column;
  height: 90vh;
  align-items: center;

  h3 {
    align-self: flex-start;
    color: $secondarycolor;
    font-family: $secondaryfont;
    font-size: 35px;
    margin: 8vh 0 0 10vw;
  }
}

.sign-up-form {
  align-items: center;
  background-color: #FFF;
  display: flex;
  flex-direction: column;
  height: 65vh;
  margin-top: 2vh;
  overflow-y: scroll;
  padding-top: 1vw;
  width: 80vw;

  label {
    align-self: flex-start;
    font-family: $primaryfont;
    font-size: 20px;
    margin: 1.5vw 0 .5vw 2.2vw;
  }

  input {
    border: 5px solid #C7D0D7;
    font-family: $primaryfont;
    font-size: 30px;
    height: 35px;
    width: 75vw;
  }

  select {
    -webkit-appearance: menulist-button;
    background-color: #FFF;
    border: 5px solid #C7D0D7;
    font-family: $primaryfont;
    width: 76vw;
  }
  
  .select-box {
    font-size: 19px;
    height: 45px;
  }

  button {
    width: 25vw;
    margin: 2vw;
    padding: 1vw;
    border: 5px solid $secondarycolor;
    color: $secondarycolor;
    font-size: 2vw;
    font-family: $secondaryfont;
    &:hover {
      background-color: $secondarycolor;
      color: white;
      cursor: pointer;
    }
  }
}
