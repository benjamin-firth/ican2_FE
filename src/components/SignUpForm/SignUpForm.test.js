import "@testing-library/jest-dom/extend-expect";
import React from 'react';
import { render, fireEvent } from "@testing-library/react";
import { shallow } from 'enzyme';
import SignUpForm from './SignUpForm';

jest.mock("react-redux", () => ({
  useSelector: () => 'mockState',
  useDispatch: () => jest.fn()
}));

describe('SignUpForm', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow( <SignUpForm />)
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('Inputs', () => {
    it('name input', () => {
      expect(wrapper.find('#stateName').prop('value')).toEqual('')
      wrapper.find('#stateName').simulate('change', {target: {value: 'test name'}})
      // wrapper.update()
      expect(wrapper.find('#stateName').prop('value')).toEqual('test name')
    });

    it('email input', () => {
      expect(wrapper.find('#stateEmail').prop('value')).toEqual('')
      wrapper.find('#stateEmail').simulate('change', {target: {value: 'test email'}})
      // wrapper.update()
      expect(wrapper.find('#stateEmail').prop('value')).toEqual('test email')
    });

    it('about input', () => {
      expect(wrapper.find('#stateAbout').prop('value')).toEqual('')
      wrapper.find('#stateAbout').simulate('change', {target: {value: 'test about me'}})
      // wrapper.update()
      expect(wrapper.find('#stateAbout').prop('value')).toEqual('test about me')
    });

    it('about input', () => {
      wrapper.find('#stateMentorBool').simulate('change', {target: {value: true}})
      expect(wrapper.find('#stateWorkDay').prop('value')).toEqual('')
      wrapper.find('#stateWorkDay').simulate('change', {target: {value: 'test work day'}})
      // wrapper.update()
      expect(wrapper.find('#stateWorkDay').prop('value')).toEqual('test work day')
    });

    it('about input', () => {
      wrapper.find('#stateMentorBool').simulate('change', {target: {value: true}})
      expect(wrapper.find('#stateEnjoyQ').prop('value')).toEqual('')
      wrapper.find('#stateEnjoyQ').simulate('change', {target: {value: 'test stateEnjoyQ'}})
      // wrapper.update()
      expect(wrapper.find('#stateEnjoyQ').prop('value')).toEqual('test stateEnjoyQ')
    });

    it('about input', () => {
      wrapper.find('#stateMentorBool').simulate('change', {target: {value: true}})
      expect(wrapper.find('#adviceQ').prop('value')).toEqual('')
      wrapper.find('#adviceQ').simulate('change', {target: {value: 'test adviceQ'}})
      // wrapper.update()
      expect(wrapper.find('#adviceQ').prop('value')).toEqual('test adviceQ')
    });

    it('about input', () => {
      wrapper.find('#stateMentorBool').simulate('change', {target: {value: true}})
      expect(wrapper.find('#stateAdviceQ').prop('value')).toEqual('')
      wrapper.find('#stateAdviceQ').simulate('change', {target: {value: 'test advice'}})
      // wrapper.update()
      expect(wrapper.find('#stateAdviceQ').prop('value')).toEqual('test advice')
    });
  });
});
