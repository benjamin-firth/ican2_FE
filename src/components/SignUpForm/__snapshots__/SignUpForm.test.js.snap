// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUpForm should match the snapshot 1`] = `
<section
  className="sign-up-container"
>
  <h3>
    build your profile
  </h3>
  <form
    className="sign-up-form"
  >
    <label>
      WHAT IS YOUR FIRST NAME?
    </label>
    <input
      id="stateName"
      onChange={[Function]}
      value=""
    />
    <label>
      WHAT IS YOUR EMAIL ADDRESS?
    </label>
    <input
      id="stateEmail"
      onChange={[Function]}
      value=""
    />
    <label>
      WHAT IS YOUR CURRENT CITY AND STATE?
    </label>
    <input
      id="countryId"
      name="country"
      type="hidden"
      value="US"
    />
    <div
      className="city-state-select"
    >
      <div
        className="select"
      >
        <select
          className="states order-alpha select-box state-select"
          id="stateId"
          name="state"
          onChange={[Function]}
        >
          <option
            value=""
          >
            Select State
          </option>
        </select>
      </div>
      <div
        className="select"
      >
        <select
          className="cities order-alpha select-box"
          id="cityId"
          name="city"
          onChange={[Function]}
        >
          <option
            value=""
          >
            Select City
          </option>
        </select>
      </div>
    </div>
    <label>
      WHAT ARE YOUR PRONOUNS?
    </label>
    <div
      className="select"
    >
      <select
        className="select-box"
        onChange={[Function]}
      >
        <option
          value=""
        >
          Select Pronouns
        </option>
        <option
          value="She/Her"
        >
          She/Her
        </option>
        <option
          value="He/Him"
        >
          He/Him
        </option>
        <option
          value="They/Them"
        >
          They/Them
        </option>
        <option
          value="none-specified"
        >
          I prefer not to say
        </option>
      </select>
    </div>
    <label>
      TELL US A LITTLE ABOUT YOURSELF (This will appear in your profile)
    </label>
    <input
      id="stateAbout"
      onChange={[Function]}
      value=""
    />
    <label>
      DO YOU WANT TO BE A MENTOR?
    </label>
    <div
      className="select"
    >
      <select
        className="select-box"
        id="stateMentorBool"
        onChange={[Function]}
        value={false}
      >
        <option
          value={false}
        >
          No
        </option>
        <option
          value={true}
        >
          Yes
        </option>
      </select>
    </div>
    <label>
      UPLOAD A PROFILE IMAGE
    </label>
    <Dropzone
      accept="image/*"
      multiple={false}
      onDrop={[Function]}
    >
      <Component />
    </Dropzone>
    <div>
      <div />
    </div>
    <button
      onClick={[Function]}
      type="button"
    >
      submit
    </button>
  </form>
</section>
`;
